packages/camera/ios/Classes/CameraPlugin.m

```diff
@@ -800,34 +802,36 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
 }
 
 - (void)handleMethodCallAsync:(FlutterMethodCall *)call result:(FlutterResult)result {
-  if ([@"availableCameras" isEqualToString:call.method]) {
-    AVCaptureDeviceDiscoverySession *discoverySession = [AVCaptureDeviceDiscoverySession
-        discoverySessionWithDeviceTypes:@[ AVCaptureDeviceTypeBuiltInWideAngleCamera ]
-                              mediaType:AVMediaTypeVideo
-                               position:AVCaptureDevicePositionUnspecified];
-    NSArray<AVCaptureDevice *> *devices = discoverySession.devices;
-    NSMutableArray<NSDictionary<NSString *, NSObject *> *> *reply =
-        [[NSMutableArray alloc] initWithCapacity:devices.count];
-    for (AVCaptureDevice *device in devices) {
-      NSString *lensFacing;
-      switch ([device position]) {
-        case AVCaptureDevicePositionBack:
-          lensFacing = @"back";
-          break;
-        case AVCaptureDevicePositionFront:
-          lensFacing = @"front";
-          break;
-        case AVCaptureDevicePositionUnspecified:
-          lensFacing = @"external";
-          break;
+  if (@available(iOS 10.0, *)) {
```

Comment:
If call.method is `availableCameras` and `@available(iOS 10.0, *)` isn't true, should it `result(@[]);`?  Or do we want the `FlutterMethodNotImplemented` (because I guess it isn't, though I don't know the side effects of that)?

Reply:
I think FlutterMethodNotImplemented is most appropriate, since we haven't implemented the method for that platform.

Today, this is a hard crash for the end user. At least this way they can handle it in application code.